consensus/src/experimental/buffer_manager.rs

```diff
@@ -1,33 +1,42 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+use std::sync::Arc;
+
+use futures::{
+    channel::{
+        mpsc::{UnboundedReceiver, UnboundedSender},
+        oneshot,
+    },
+    executor::block_on,
+    SinkExt,
+};
+
+use consensus_types::{common::Author, executed_block::ExecutedBlock};
+use diem_crypto::ed25519::Ed25519Signature;
+use diem_types::{
+    account_address::AccountAddress,
+    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+    validator_verifier::ValidatorVerifier,
+};
+
 use crate::{
     experimental::{
         buffer_item::BufferItem,
         execution_phase::{ExecutionRequest, ExecutionResponse},
-        linkedlist::{get_elem, get_next, link_eq, set_elem, take_elem, Link, List},
+        linkedlist::{get_elem, get_elem_mut, get_next, link_eq, set_elem, take_elem, Link, List},
         persisting_phase::{PersistingRequest, PersistingResponse},
         signing_phase::{SigningRequest, SigningResponse},
     },
     network::NetworkSender,
+    network_interface::ConsensusMsg,
     round_manager::VerifiedEvent,
     state_replication::StateComputerCommitCallBackType,
 };
-use consensus_types::{common::Author, executed_block::ExecutedBlock};
-use diem_types::{
-    account_address::AccountAddress,
-    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
-    validator_verifier::ValidatorVerifier,
-};
-use futures::{
-    channel::{
-        mpsc::{UnboundedReceiver, UnboundedSender},
-        oneshot,
-    },
-    SinkExt,
-};
+use std::intrinsics::unreachable;
```

Comment:
hmm I don't remember the macro needs import

Reply:
this line is added by the IDE.. I just removed it.